src/test/java/reposense/parser/CsvParserTest.java

```diff
@@ -131,14 +151,72 @@ public void merge_twoRepoConfigs_success() throws ParseException, IOException {
     }
 
+    @Test
+    public void merge_emptyLocation_success() throws ParseException, IOException {
+        FIRST_AUTHOR.setIgnoreGlobList(FIRST_AUTHOR_GLOB_LIST);
+        SECOND_AUTHOR.setIgnoreGlobList(REPO_LEVEL_GLOB_LIST);
+
+        List<Author> expectedBetaAuthors = new ArrayList<>();
+        expectedBetaAuthors.add(FIRST_AUTHOR);
+        expectedBetaAuthors.add(SECOND_AUTHOR);
+
+        List<Author> expectedDeltaAuthors = new ArrayList<>();
+        expectedDeltaAuthors.add(FIRST_AUTHOR);
+
+        RepoConfiguration expectedBetaConfig =
+                new RepoConfiguration(new RepoLocation(TEST_REPO_BETA_LOCATION), TEST_REPO_BETA_BRANCH);
+        expectedBetaConfig.setAuthorList(expectedBetaAuthors);
+        expectedBetaConfig.setAuthorDisplayName(FIRST_AUTHOR, "Nbr");
+        expectedBetaConfig.setAuthorDisplayName(SECOND_AUTHOR, "Zac");
+        expectedBetaConfig.addAuthorAliases(SECOND_AUTHOR,  Arrays.asList("Zachary Tang"));
+        expectedBetaConfig.setIgnoreGlobList(REPO_LEVEL_GLOB_LIST);
+
+        RepoConfiguration expectedDeltaConfig =
+                new RepoConfiguration(new RepoLocation(TEST_REPO_DELTA_LOCATION), TEST_REPO_DELTA_BRANCH);
+        expectedDeltaConfig.setAuthorList(expectedDeltaAuthors);
+        expectedDeltaConfig.setAuthorDisplayName(FIRST_AUTHOR, "Nbr");
+
+        List<RepoConfiguration> expectedConfigs = new ArrayList<>();
+        expectedConfigs.add(expectedBetaConfig);
+        expectedConfigs.add(expectedDeltaConfig);
+
+        String input = String.format("-config %s", MERGE_EMPTY_LOCATION_FOLDER);
+        CliArguments cliArguments = ArgsParser.parse(translateCommandline(input));
+
+        List<RepoConfiguration> actualConfigs =
+                new RepoConfigCsvParser(((ConfigCliArguments) cliArguments).getRepoConfigFilePath()).parse();
+        List<RepoConfiguration> authorConfigs =
+                new AuthorConfigCsvParser(((ConfigCliArguments) cliArguments).getAuthorConfigFilePath()).parse();
+        RepoConfiguration.merge(actualConfigs, authorConfigs);
+
+        Assert.assertEquals(2, actualConfigs.size());
+        Assert.assertEquals(expectedConfigs, actualConfigs);
+
+        RepoConfiguration firstExpectedConfig = expectedConfigs.get(0);
+        RepoConfiguration secondExpectedConfig = expectedConfigs.get(1);
+        RepoConfiguration firstActualConfig = actualConfigs.get(0);
+        RepoConfiguration secondActualConfig = actualConfigs.get(1);
+
+        Assert.assertEquals(firstExpectedConfig.getLocation(), firstActualConfig.getLocation());
+        Assert.assertEquals(firstExpectedConfig.getAuthorList(), firstActualConfig.getAuthorList());
+        Assert.assertEquals(firstExpectedConfig.getAuthorDisplayNameMap(), firstActualConfig.getAuthorDisplayNameMap());
+        Assert.assertEquals(firstExpectedConfig.getAuthorAliasMap(), firstActualConfig.getAuthorAliasMap());
+        Assert.assertEquals(REPO_LEVEL_GLOB_LIST, firstActualConfig.getIgnoreGlobList());
```

Comment:
Actually, `TestUtil` has this helper method `compareRepoConfig` already which seems to be doing the same thing(before you changed from `hashCode`) here. Maybe try using that method instead, makes the code alot cleaner as well.

Reply:
Should I change the `compareRepoConfig` method in `TestUtil` as well? Because it also contains unnecessary `hashcode`.

Reply:
Or, we can leave it in this way and change all the unnecessary compare of RepoConfig in all tests in a seperate PR? Because there are many other tests that do not use `compareRepoConfig` method.

Reply:
Let's change the `TestUtil#compareRepofConfig` method to what @yong24s has mentioned here, but don't need to touch all existing tests to use that, just ensure the new test methods you have implemented use that instead.

We can open a new issue for the other test methods to use the helper method.